stacks_queues/queue.py

```diff
@@ -23,39 +48,97 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+        if ((self.rear +1) % INITIAL_QUEUE_SIZE ==self.front):
+            raise QueueFullException('This will break it')
+
+        elif (self.front == -1): 
+            self.front = 0
+            self.rear = 0
+            self.store[self.rear] = element
+            self.size = self.size + 1
+        else:
+             
+            # next position of rear
+            self.rear = (self.rear + 1) % INITIAL_QUEUE_SIZE
+            self.store[self.rear] = element
+            self.size = self.size + 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
-            The Queue is empty.
+            The Queue is empty. In the circular Queue it remove the item at the front 
         """
-        pass
+        
+        if self.empty():
+            raise QueueEmptyException("The Stack is empty")
+
+        # condition for only one element
+        elif (self.front == self.rear):
+            tmp=self.store[self.front]
+            self.front = -1
+            self.rear = -1
+            self.size -=1
+            return tmp
+
+        else:
+            tmp = self.store[self.front]
+            self.front = (self.front + 1) % INITIAL_QUEUE_SIZE
+            self.size -=1
+            return tmp
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.size == 0:
+            return None
+        return self.store[self.front]
         
-
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
-
+        return self.size
+        # self if a reference to the current class when in a class method you need to append self to it.  
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        if self.size == 0:
+            return True
+        return False
```

Comment:
This is a bit simpler

```suggestion
        return self.size == 0
```